### PR TITLE
Bug channel annot merge

### PR DESCRIPTION
**Pull request by nmarkowitz**
#### Reference issue

[Example: Fixes #1234.](https://github.com/mne-tools/mne-qt-browser/issues/267)

#### What does this implement/fix?

Fix issues when combining channel-based annotations with another annotation.

#### Additional information

Channel-based annotation can not merge with any other annotation.

### CODE REVIEW

**Review by mscheltienne — COMMENTED** [review]
Preventing merge between 2 annotations if at least one has a channe-specific annotation works for me. That's a good rule to start with and it can be changed later. 

I am however greeted with this error when dragging a channel-specific annotation inside another one:

```
Traceback (most recent call last):
  File "/home/scheltie/git/mscheltienne/mne-qt-browser/mne_qt_browser/_pg_figure.py", line 4645, in _region_changed
    idx = self._get_onset_idx(region.old_onset)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scheltie/git/mscheltienne/mne-qt-browser/mne_qt_browser/_pg_figure.py", line 4639, in _get_onset_idx
    idx = np.where(self.mne.inst.annotations.onset == onset)[0][0]
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: index 0 is out of bounds for axis 0 with size 0
```

[screncast-issue.webm](https://github.com/user-attachments/assets/735c4857-4147-4889-bb10-795ecee07d53)

**Comment by nmarkowitz** [comment]
@mscheltienne I just checked and this is a bug even for general annotations. I can fix this in this or another PR.

https://github.com/user-attachments/assets/897dd456-f573-425f-959f-eb2a4bd31b6e

**Comment by larsoner** [comment]
Feel free to fix in this PR

File: mne_qt_browser/_pg_figure.py [hunk 1]
```diff
@@ -2435,29 +2436,55 @@ def __init__(self, mne, description, values, weakmain, ch_names=None):
...
+
         onset = np.min(regions_[:, 0])
         offset = np.max(regions_[:, 1])
+
+        self.old_onset = onset
+
         logger.debug(f"New {self.description} region: {onset:.2f} - {offset:.2f}")
         # remove overlapping regions
         for region in overlapping_regions:
+            # region.removeSingleChannelAnnots.emit(region)
```
Comment (larsoner) [inline mne_qt_browser/_pg_figure.py:2482]: Cruft?

**Review by larsoner — COMMENTED** [review]
@nmarkowitz can you verify that this works properly, and then add unit tests? Then we should be able to merge!

File: mne_qt_browser/tests/test_pg_specific.py [hunk 2]
```diff
@@ -167,6 +189,17 @@ def test_ch_specific_annot(raw_orig, pg_backend):
...
+
+        # Check that channel specific annotations do not merge
+        fig._fake_click(
+            (2.0, 1.0), add_points=[(3.0, 1.0)], xform="data", button=1, kind="drag"
+        )
+        with pytest.warns(RuntimeWarning) as record:
+            fig._fake_click(
+                (2.1, 1.0), add_points=[(5.0, 1.0)], xform="data", button=1, kind="drag"
+            )
+        assert len(record) == 1
```
Comment (mscheltienne) [inline mne_qt_browser/tests/test_pg_specific.py:201]: Could you remove `record` in favor of `with pytest.warns(RuntimeWarning, match="...")`? 
This way we actually capture only `RuntimeWarning` with a message which match the regex pattern provided. Something like: `with.pytest.warns(RuntimeWarning, match="Can not combine channel-based")`.

**Review by mscheltienne — APPROVED** [review]
I tried to broke it again, but failed! Looks fixed to me, thanks a lot!

File: mne_qt_browser/tests/test_pg_specific.py (hunk 2 above)
Comment (nmarkowitz) [reply mne_qt_browser/tests/test_pg_specific.py]: done

**Comment by nmarkowitz** [comment]
@larsoner this one is ready to be merged

**Review by larsoner — APPROVED** [review]
Looks good, thanks @nmarkowitz !

**Pull request merged**